File: Problems/No.55_Jump_Game/jump_game.py
class Solution(object):
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """

        # parameters
        Mstep = 0
        sizeN = len(nums)

        # start computing
        for i in range(sizeN):
            if ((i > Mstep) or (Mstep >= sizeN-1)):
                break
            else:
                Mstep = max(Mstep, i+nums[i])
        
        return True if (Mstep >= sizeN-1) else False

File: Problems/No.55_Jump_Game/jump_game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        
        // parameters
        int Mstep = 0;
        int sizeN = nums.size();

        // start computing
        for (int i=0; i<sizeN; i++) {

            if ((i > Mstep) || (Mstep >= sizeN-1)) break;
            else Mstep = std::max(Mstep, i+nums[i]);

        }

        return true ? (Mstep >= sizeN-1) : false;

    }
};
